### PR TITLE
Fix branch tooltip 

### LINKED ISSUE #12008: Hover tooltip of branch name not showing anymore
### Describe the bug

With a recent update, hovering over a branch name with the mouse does not show the branch name anymore.
In the past this was showing. 

The problem is when we have pretty long branch names, where the interesting part is at the end e.g.

- ticketID-verylong-branch-name
- ticketID-verylong-branch-name-bugfix1
- ticketID-verylong-branch-name-bugfix2

So if I am interested in e.g. in bugfix2, I just see the beginning of the name and need to try them or guess by the (12 hours ago) timestamp.

### Version & OS

Github Desktop Version 2.7.2
MacOSX 11.2.3 (Big Sur)

### Steps to reproduce the behavior

1. Open _Current branch_ dropdown
2. hover over any branch name with the mouse and wait 2-4s over each item (don't move the mouse)

### Expected behavior

An HTML tooltip (usually as from the HTML title attribute) should appear. 

### Actual behavior

Tooltip does not appear.

### Screenshots

image

The screenshot shows how it is looking at the moment without the hover tooltip appearing. Unfortunately I don't have a screenshot of a version where it was still working. 

But the tooltip is showing in _merge other branch into current branch_ modal:

image

That's how it should be.

### Logs

-

### Additional context

-

### PR DESCRIPTION
**Pull request by tidy-dev**
Closes #12008

## Description

Add tooltip back when hovering over branch name in the branch-dropdown.

Also add "Copy to branch name" to macOS cherry-pick operation since tool tip is disabled during cherry-picking due to cursor conflicts.

### Screenshots
** UPDATE from other macOS video -> copy to message to feel like a tool tip on mac **

https://user-images.githubusercontent.com/75402236/115758347-04124100-a36d-11eb-8bd5-9dd9cf87c8f4.mov

MacOS

https://user-images.githubusercontent.com/75402236/115245676-3153ba80-a0f3-11eb-81c2-66da83bfb3d7.mov

Windows

https://user-images.githubusercontent.com/75402236/115246601-087ff500-a0f4-11eb-8746-2c208e129abc.mov

## Release notes

Notes: [Fixed] Show tooltip with full branch name in branch dropdown.

### CODE REVIEW

**Review by billygriffin — DISMISSED** [review]
💯

File: app/styles/ui/drag-elements/_cherry-pick-commit.scss [hunk 1]
```diff
@@ -55,6 +55,7 @@
       display: flex;
       align-items: center;
       justify-content: center;
+      white-space: nowrap;
```
Comment (sergiou87) [inline app/styles/ui/drag-elements/_cherry-pick-commit.scss:72]: Why did you need this?

File: app/styles/ui/_app.scss [hunk 2]
```diff
@@ -45,6 +45,12 @@
             cursor: default;
           }
 
+          .branches-list-item {
+            div {
+              pointer-events: none;
+            }
+          }
```
Comment (sergiou87) [inline app/styles/ui/_app.scss:52]: Why did you need this?

**Review by sergiou87 — DISMISSED** [review]
LGTM! And I like the result 😄 

I just wonder why some of the included changes are needed 😳  Other than that :shipit:

File: app/styles/ui/drag-elements/_cherry-pick-commit.scss (hunk 1 above)
Comment (tidy-dev) [reply app/styles/ui/drag-elements/_cherry-pick-commit.scss:72]: image
Other wise got weird wrapping.. sometimes even the Copy To would split to two lines... and this was current problem with windows if you had a really long branch name in comparison to the commit message.

File: app/styles/ui/_app.scss (hunk 2 above)
Comment (tidy-dev) [reply app/styles/ui/_app.scss:52]: Otherwise.. the cursor will not show up when hovering over it. Thus, during cherry-pick when moving your mouse over the branch-list-item you will see the copy curser for a second (when not actually on the branch name or time) then it will disappear when you go over the branch-name and then reappear when you leave the branch-name. 

Also.. this prevents the tool tip that we don't want when during cherry-picking.

File: app/styles/ui/drag-elements/_cherry-pick-commit.scss [hunk 3]
```diff
@@ -16,9 +16,11 @@
     flex-direction: row;
     align-items: center;
     background-color: var(--background-color);
+    overflow: hidden;
 
     .commit {
       border-bottom: none;
+      max-width: 300px;
```
Comment (tidy-dev) [inline app/styles/ui/drag-elements/_cherry-pick-commit.scss:23]: Not related to this PR: Just noticed that the commit ghost will expand to show entire commit message which is weird looking for really long commit messages.

**Review by sergiou87 — APPROVED** [review]
LGTM! The result looks great 😍
